lib/src/default_revset_engine.rs

```diff
@@ -99,6 +99,16 @@ impl<'index> Revset<'index> for RevsetImpl<'index> {
     }
 }
 
+struct RevsetCommitIdIterator<I>(I);
+
+impl<'index, I: Iterator<Item = IndexEntry<'index>>> Iterator for RevsetCommitIdIterator<I> {
+    type Item = CommitId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next().map(|index_entry| index_entry.commit_id())
```

Comment:
nit: this can be inlined as `self.inner.iter().map(...)` (if we won't add more complexity to this iterator.)

Reply:
Why didn't I think of that? :) Done.

Reply:
Oh, not actually done because I apparently clicked "Resolve" before the push finished :( I'll send a PR to fix